contracts/src/spaces/facets/prepay/PrepayFacet.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+// interfaces
+import {IPrepay} from "contracts/src/spaces/facets/prepay/IPrepay.sol";
+import {IPlatformRequirements} from "contracts/src/factory/facets/platform/requirements/IPlatformRequirements.sol";
+
+// libraries
+import {MembershipStorage} from "contracts/src/spaces/facets/membership/MembershipStorage.sol";
+import {CurrencyTransfer} from "contracts/src/utils/libraries/CurrencyTransfer.sol";
+
+// contracts
+import {PrepayBase} from "./PrepayBase.sol";
+import {ReentrancyGuard} from "contracts/src/diamond/facets/reentrancy/ReentrancyGuard.sol";
+import {Entitled} from "contracts/src/spaces/facets/Entitled.sol";
+import {Facet} from "contracts/src/diamond/facets/Facet.sol";
+
+contract PrepayFacet is IPrepay, PrepayBase, ReentrancyGuard, Entitled, Facet {
+  function __PrepayFacet_init() external onlyInitializing {
+    _addInterface(type(IPrepay).interfaceId);
+  }
+
+  function prepayMembership(
+    uint256 supply
+  ) external payable nonReentrant onlyOwner {
+    if (supply == 0) revert Prepay__InvalidSupplyAmount();
+
+    MembershipStorage.Layout storage ds = MembershipStorage.layout();
+    IPlatformRequirements platform = IPlatformRequirements(ds.spaceFactory);
+
+    uint256 cost = supply * platform.getMembershipFee();
```

Comment:
why not using calculateMembershipPrepayFee function defined below? if not why keep that?

Reply:
good catch! will update.